ow_plexil/scripts/plexilcow

```diff
@@ -0,0 +1,27 @@
+#!/bin/bash
+
+# plexilcow (PLEXIL Compiler for OceanWATERS).
+
+# Simple script for compiling a single PLEXIL plan (.plp file).
+# Compiled version (.plx) is placed in the current deployment
+# directory.  Assumes that the PLEXIL compiler (plexilc) is in the
+# user's search path.
+
+if [ $# -ne 1 ]; then
+    echo "Usage: $0 <filename.plp>"
+    exit 1
+fi
+
+input_file="$1"
+output_directory="$HOME/ow/devel/.private/ow_plexil/etc/plexil/"
```

Comment:
This assumes the users workspace is in their home directory, which is not the case for me. 

I believe so long as devel/setup.bash has been sourced there is some kind of "ROS workspace home" variable we could use here. Will do some digging.

Reply:
Good catch!

Reply:
I think I found something. Unfortunately it could be considered slightly hacky, but I think it's relatively reliable. 
```suggestion
devel_path=$(echo $CMAKE_PREFIX_PATH | cut -d ":" -f 1)
output_directory="$devel_path/.private/ow_plexil/etc/plexil/"
```

Reply:
There is `PLEXIL_PLAN_DIR` that gets you to `devel/etc/plexil` but plans are compiled into a `.private` tree, and I haven't found a direct way to that.

One solution, that seems reasonable to me, is introducing a workspace home variable for OW and requiring it to use this script...

Reply:
> One solution, that seems reasonable to me, is introducing a workspace home variable for OW and requiring it to use this script...

Apparently, but i don't think said variable can remain internal. No need to export it unless something else requires it. I am surprised to learn that ROS or WS don't define exactly that already.

Reply:
I think something like `$PLEXIL_PLAN_DIR/../..` is a perfectly fine solution. Up to you which you use.

Reply:
Our comments crossed.  Your suggestion is clever but not only hacky, it may not work if the user has more than one ROS workspace.

I looked at many web pages and am giving up my search for a built-in solution.

What do you mean by a variable internal to the script?

Reply:
> I think something like `$PLEXIL_PLAN_DIR/../..` is a perfectly fine solution. Up to you which you use.

Let's go with this one -- I think it's a fine start anyway.